swyft/store/dataset.py

```diff
@@ -10,104 +10,125 @@
 
 
 class Dataset(torch_Dataset):
-    """Dataset for access to swyft.Store."""
-
-    def __init__(self, N, prior, store, bound=None, simhook=None, simkeys = None):
-        """Initialize Dataset.
-
-        Args:
-            N (int): Number of samples.
-            prior (swyft.Prior): Parameter prior.
-            store (swyft.Store): Store reference.
-            simhook (Callable): Posthook for simulations. Applied on-the-fly to each point.
-            simkeys (list of strings): List of simulation keys that should be exposed 
-                                        (None means that all store sims are exposed).
-
-        Notes:
-            Due to the statistical nature of the Store, the returned number of
-            samples is effectively drawn from a Poisson distribution with mean
-            N.
-        """
+    """Dataset for access to swyft.Store.
+
+    Args:
+        N (int): Number of samples.
+        prior (swyft.Prior): Parameter prior.
+        store (swyft.Store): Store reference.
+        simhook (Callable): Posthook for simulations. Applied on-the-fly to each point.
+        simkeys (list of strings): List of simulation keys that should be exposed 
+                                    (None means that all store sims are exposed).
+
+    .. note::
+        swyft.Dataset is essentially a list of indices that point to
+        corresponding entries in the swyft.Store.  It is a daugther class of
+        torch.utils.data.Dataset, and can be used directly for training.  Due
+        to the statistical nature of the Store, the returned number of samples
+        is effectively drawn from a Poisson distribution with mean N.  
+    """
+
+    def __init__(self, N, prior, store, bound=None, simhook=None, simkeys=None):
         super().__init__()
 
         # Initialization
         self._trunc_prior = swyft.TruncatedPrior(prior, bound)
         self._indices = store.sample(N, prior, bound=bound)
+        if len(self._indices) == 0:
+            raise RuntimeError("Not enough simulations in store")
 
         self._store = store
         self._simhook = simhook
         self._simkeys = simkeys if simkeys else list(self._store.sims)
+        self._pnames = self._store.pnames
+
+        if self.requires_sim:
+            print("WARNING: Some points require simulation.")
 
     def __len__(self):
+        """Return length of dataset."""
         return len(self._indices)
 
     @property
     def prior(self):
+        """Return prior of dataset (swyft.Prior)."""
         return self._trunc_prior.prior
 
     @property
     def bound(self):
+        """Return bound of truncated prior of dataset (swyft.Bound)."""
         return self._trunc_prior.bound
 
     def _tensorfy(self, x):
         return {k: torch.tensor(v).float() for k, v in x.items()}
```

Comment:
[ ] we should use the function `array_to_tensor` whenever we convert from x to tensor.